src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -312,36 +338,56 @@ private IConfidentialClientApplication BuildConfidentialClientApplication()
                 request.PathBase,
                 microsoftIdentityOptions.CallbackPath.Value ?? string.Empty);
 
-            if (!applicationOptions.Instance.EndsWith("/"))
-                applicationOptions.Instance += "/";
-
-            string authority ;
-            IConfidentialClientApplication app = null;
-
-            if (microsoftIdentityOptions.IsB2C)
+           if (string.IsNullOrEmpty(applicationOptions.Instance) &&
+                !string.IsNullOrEmpty(microsoftIdentityOptions.Instance))
```

Comment:
This needs to be investigated. I found that the applicationOptions did not inherit from the microsoftIdentityOptions in this case, so I had to add values for both in the integration test. Issue tracked here: [https://github.com/AzureAD/microsoft-identity-web/issues/37](https://github.com/AzureAD/microsoft-identity-web/issues/37)

Reply:
separate PR and by fixing the test set-up, i no longer have the issue, but we should still look into merging the class. will update the issue.

---
In reply to: [385444897](https://github.com/AzureAD/microsoft-identity-web/pull/39#discussion_r385444897) [](ancestors = 385444897)